include/nuttx/pci/pcie_designware.h

```diff
@@ -0,0 +1,567 @@
+   __dw_pcie_ver_cmp(_pci, TYPE_ ## _type, ==))
+
+#define dw_pcie_ver_type_is_ge(_pci, _ver, _type) \
+  (__dw_pcie_ver_cmp(_pci, _ver, ==) && \
+   __dw_pcie_ver_cmp(_pci, TYPE_ ## _type, >=))
+
+/* DWC PCIe controller capabilities */
+#define DW_PCIE_CAP_REQ_RES   0
+#define DW_PCIE_CAP_IATU_UNROLL   1
+#define DW_PCIE_CAP_CDM_CHECK   2
+
+#define dw_pcie_cap_is(_pci, _cap) \
+  test_bit(DW_PCIE_CAP_ ## _cap, &(_pci)->caps)
+
+#define dw_pcie_cap_set(_pci, _cap) \
+  set_bit(DW_PCIE_CAP_ ## _cap, &(_pci)->caps)
+
+/* Parameters for the waiting for link up routine */
+#define LINK_WAIT_MAX_RETRIES   10
+#define LINK_WAIT_SLEEP_MS    90
+
+/* Parameters for the waiting for iATU enabled routine */
+#define LINK_WAIT_MAX_IATU_RETRIES  5
+#define LINK_WAIT_IATU      9
+
+/* Synopsys-specific PCIe configuration registers */
+#define PCIE_PORT_FORCE     0x708
+#define PORT_FORCE_DO_DESKEW_FOR_SRIS BIT(23)
+
+#define PCIE_PORT_AFR     0x70C
+#define PORT_AFR_N_FTS_MASK   GENMASK(15, 8)
+#define PORT_AFR_N_FTS(n)   FIELD_PREP(PORT_AFR_N_FTS_MASK, n)
+#define PORT_AFR_CC_N_FTS_MASK    GENMASK(23, 16)
+#define PORT_AFR_CC_N_FTS(n)    FIELD_PREP(PORT_AFR_CC_N_FTS_MASK, n)
+#define PORT_AFR_ENTER_ASPM   BIT(30)
+#define PORT_AFR_L0S_ENTRANCE_LAT_SHIFT 24
+#define PORT_AFR_L0S_ENTRANCE_LAT_MASK  GENMASK(26, 24)
+#define PORT_AFR_L1_ENTRANCE_LAT_SHIFT  27
+#define PORT_AFR_L1_ENTRANCE_LAT_MASK GENMASK(29, 27)
+
+#define PCIE_PORT_LINK_CONTROL    0x710
+#define PORT_LINK_DLL_LINK_EN   BIT(5)
+#define PORT_LINK_FAST_LINK_MODE  BIT(7)
+#define PORT_LINK_MODE_MASK   GENMASK(21, 16)
+#define PORT_LINK_MODE(n)   FIELD_PREP(PORT_LINK_MODE_MASK, n)
+#define PORT_LINK_MODE_1_LANES    PORT_LINK_MODE(0x1)
+#define PORT_LINK_MODE_2_LANES    PORT_LINK_MODE(0x3)
+#define PORT_LINK_MODE_4_LANES    PORT_LINK_MODE(0x7)
+#define PORT_LINK_MODE_8_LANES    PORT_LINK_MODE(0xf)
+
+#define PCIE_PORT_LANE_SKEW   0x714
+#define PORT_LANE_SKEW_INSERT_MASK  GENMASK(23, 0)
+
+#define PCIE_PORT_DEBUG0    0x728
+#define PORT_LOGIC_LTSSM_STATE_MASK 0x1f
+#define PORT_LOGIC_LTSSM_STATE_L0 0x11
+#define PCIE_PORT_DEBUG1    0x72C
+#define PCIE_PORT_DEBUG1_LINK_UP    BIT(4)
+#define PCIE_PORT_DEBUG1_LINK_IN_TRAINING BIT(29)
+
+#define PCIE_LINK_WIDTH_SPEED_CONTROL 0x80C
+#define PORT_LOGIC_N_FTS_MASK   GENMASK(7, 0)
+#define PORT_LOGIC_SPEED_CHANGE   BIT(17)
+#define PORT_LOGIC_LINK_WIDTH_MASK  GENMASK(12, 8)
+#define PORT_LOGIC_LINK_WIDTH(n)  FIELD_PREP(PORT_LOGIC_LINK_WIDTH_MASK, n)
+#define PORT_LOGIC_LINK_WIDTH_1_LANES PORT_LOGIC_LINK_WIDTH(0x1)
+#define PORT_LOGIC_LINK_WIDTH_2_LANES PORT_LOGIC_LINK_WIDTH(0x2)
+#define PORT_LOGIC_LINK_WIDTH_4_LANES PORT_LOGIC_LINK_WIDTH(0x4)
+#define PORT_LOGIC_LINK_WIDTH_8_LANES PORT_LOGIC_LINK_WIDTH(0x8)
+
+#define PCIE_MSI_ADDR_LO    0x820
+#define PCIE_MSI_ADDR_HI    0x824
+#define PCIE_MSI_INTR0_ENABLE   0x828
+#define PCIE_MSI_INTR0_MASK   0x82C
+#define PCIE_MSI_INTR0_STATUS   0x830
+
+#define GEN3_RELATED_OFF      0x890
```

Comment:
let's align the value to the same column, let's move the macro which doesn't need for chip driver to drivers/pci/pcie_dw.h